db/model/Gdoc/archieToEnriched.ts

```diff
@@ -121,28 +121,39 @@ function generateStickyNav(
 }
 
 function generateToc(
-    body: OwidEnrichedGdocBlock[]
+    body: OwidEnrichedGdocBlock[],
+    // For linear topic pages, we record h1s & h2s
+    // For the sdg-toc, we record h2s & h3s (as it was developed before we decided to use h1s as our top level heading)
+    { primary, secondary }: { primary: number; secondary: number }
 ): TocHeadingWithTitleSupertitle[] {
     const toc: TocHeadingWithTitleSupertitle[] = []
 
-    // track h2s and h3s for the SDG table of contents
     body.forEach((block) =>
         traverseEnrichedBlocks(block, (child) => {
             if (child.type === "heading") {
                 const { level, text, supertitle } = child
                 const titleString = spansToSimpleString(text)
-                const supertitleString =
-                    supertitle && spansToSimpleString(supertitle)
-                if (titleString && (level === 2 || level === 3)) {
+                const supertitleString = supertitle
+                    ? spansToSimpleString(supertitle)
+                    : ""
+                if (titleString && (level === primary || level === secondary)) {
                     toc.push({
                         title: titleString,
                         supertitle: supertitleString,
                         text: titleString,
                         slug: urlSlug(`${supertitleString} ${titleString}`),
```

Comment:
I came across two headings with the same title/slug in time-use ("Additional information"), which made them both light up at the same time in the ToC. We had a [provision in WP code to handle this special case](https://github.com/owid/owid-grapher/blob/1b41a969a955d15414872b7a0e5e357912f68508/baker/formatWordpressPost.tsx#L528-L530) but it hasn't been ported over yet. It is not the most critical bug but it can be quite visible so I'm a bit on the fence as to whether to make the fix par of this PR.

Here is an example doc: https://docs.google.com/document/d/1uYZEcCP2shwa5U1EYaGb6cRvLFEz0MrUJv9m8UASbqs/edit